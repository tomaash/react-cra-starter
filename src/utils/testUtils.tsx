import React from 'react'
import ReactDOMServer from 'react-dom/server'
const path = require('path')

export const squashHtmlTags = str => str.replace(/(<[^>]+>)+/g, '\n')

export const renderText = element =>
  squashHtmlTags(ReactDOMServer.renderToStaticMarkup(element))

export const startPolly = dirname => {
  const { Polly } = require('@pollyjs/core')
  const { setupPolly } = require('setup-polly-jest')
  const FSPersister = require('@pollyjs/persister-fs')
  const XHRAdapter = require('@pollyjs/adapter-xhr')

  Polly.register(XHRAdapter)
  Polly.register(FSPersister)

  setupPolly({
    adapters: ['xhr'],
    recordIfMissing: true,
    persister: 'fs',
    persisterOptions: {
      fs: {
        recordingsDir: path.resolve(dirname, '__recordings__')
      }
    }
  })
}

export const sleep = m => new Promise(r => setTimeout(r, m))
