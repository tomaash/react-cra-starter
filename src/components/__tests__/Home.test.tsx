import React from 'react'
import { shallow } from 'enzyme'
import { Home } from '../Home'
import { renderText, squashHtmlTags } from '~/utils/testUtils'

describe('<Home />', () => {
  it('renders component', () => {
    const text = renderText(<Home />)
    expect(text).toMatchSnapshot()
  })
})
