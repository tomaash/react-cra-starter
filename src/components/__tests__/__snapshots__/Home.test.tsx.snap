// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders component 1`] = `
"
Edit 
src/app.tsx
 and save to reload.
Learn React
"
`;
